### PR TITLE
New search implementation

### PR DESCRIPTION
**Pull request by rajatbarman**
https://github.com/tailcallhq/tailcallhq.github.io/assets/68900436/e7bcc44a-9fd6-4351-b97c-29645bfcd22a

### CODE REVIEW

File: src/css/custom.css [hunk 1]
```diff
@@ -322,23 +322,28 @@ aside.theme-doc-sidebar-container {
   margin-top: 0;
 }
 
+.algolia-docsearch-suggestion--category-header {
+  background-color: var(--ifm-color-gray-300) !important;
```
Comment (amitksingh1490) [inline src/css/custom.css:326]: gray should not be used in the name you can name it, base-primary or base-secondary

File: src/theme/DocRoot/Layout/Sidebar/Sidebar.module.css [hunk 2]
```diff
@@ -0,0 +1,81 @@
+.input {
+  width: 100%;
+  border-radius: 8px;
+  border: 1px solid var(--ifm-color-gray-800);
```
Comment (amitksingh1490) [inline src/theme/DocRoot/Layout/Sidebar/Sidebar.module.css:4]: gray used again

File: src/theme/DocRoot/Layout/Sidebar/Sidebar.module.css [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  background: transparent var(--ifm-navbar-search-input-icon) no-repeat .75rem center/1rem 1rem;
+}
+
+.inputContainer {
+  padding: 1.3rem;
+  padding-bottom: 1rem;
+}
+
+.overlay {
+  background-color: rgba(255,255,255,.3);
```
Comment (amitksingh1490) [inline src/theme/DocRoot/Layout/Sidebar/Sidebar.module.css:23]: hard coded color

File: src/theme/DocRoot/Layout/Sidebar/Sidebar.module.css [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  height: calc(100% - 44px);
+}
+
+.footer {
+  padding: .75rem 1rem;
+  border-top: 1px solid var(--ifm-color-gray-600);
```
Comment (amitksingh1490) [inline src/theme/DocRoot/Layout/Sidebar/Sidebar.module.css:57]: gray again

File: src/css/custom.css (hunk 1 above)
Comment (rajatbarman) [reply src/css/custom.css:326]: Hey, these are pre-defined docusauras variables.
https://github.com/facebook/docusaurus/issues/3955#issuecomment-1521944593

**Review by amitksingh1490 — APPROVED** [review]

**Pull request merged**